### ISSUE TITLE
cmpxchg in lkmm.h

### ISSUE THREAD

**Issue by db7** [MPI-SWS/genmc#20]
The `cmpxchg`  in [lkmm.h](https://github.com/MPI-SWS/genmc/blob/1142c61028eea365c01a2d370210b6350ce2452f/include/lkmm.h#L123) does not have the same signature as in Linux.

According to [this file](https://github.com/torvalds/linux/blob/master/arch/x86/include/asm/cmpxchg.h) and to its usage (like in the [mcs lock](https://github.com/torvalds/linux/blob/16b3d0cf5bad844daaf436ad2e9061de0fe36e5c/kernel/locking/mcs_spinlock.h#L110) ), the `cmpxchg` should return the old/new value instead of a bool as in [lkmm.h](https://github.com/MPI-SWS/genmc/blob/1142c61028eea365c01a2d370210b6350ce2452f/include/lkmm.h#L123) and in C11. That is probably a bit tricky to make as a macro, though.
 
Besides that, the `cmpxchg_{acquire, release}` in lkmm.h have wrong arguments (x instead of p).

We are not actually using LKMM, so this is not super important, but I just wanted keep record.

**Comment by db7** [comment]
Perhaps the following could work?
```c
#define __cmpxchg(p, o, n, s, f)					  \
({									  \
	__typeof__((o)) _o_ = (o);					  \
	(atomic_compare_exchange_strong_explicit(p, &_o_, n, s, f), _o_); \
})
```

**Comment by michaliskok** [comment]
Fixed in v0.6.1.
